keras/src/layers/layer.py

```diff
@@ -538,7 +541,9 @@ def add_weight(
                 initializer = "zeros"
         initializer = initializers.get(initializer)
         with backend.name_scope(self.name, caller=self):
+            value = initializer(shape, dtype)
```

Comment:
Oh, you can't change this. With the Tensorflow backend, in some cases, we need to delay the call to initializers.

Why is this needed?